test/new-e2e/language-detection/language_detection_test.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+)
+
+// testLocally should be set to true if running the test from a developer machine (i.e. not CI)
+// This sets the test suite to use the hardcoded agent version `versionStr`
+const testLocally = false
+
+const versionStr = "7.48.0~rc.1-1"
+
+const configStr = `
+process_config:
+  process_collection:
+    enabled: true
+  intervals:
+    process: 1
+language_detection:
+  enabled: true
+`
+
+type languageDetectionSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestLanguageDetectionSuite(t *testing.T) {
+	agentParams := []func(*agentparams.Params) error{
+		agentparams.WithAgentConfig(configStr),
+	}
+
+	if testLocally {
+		agentParams = append(agentParams, agentparams.WithVersion(versionStr))
+	}
+
+	e2e.Run(t, &languageDetectionSuite{}, e2e.AgentStackDef(nil, agentParams...))
+}
+
+func (s *languageDetectionSuite) checkDetectedLanguage(command string, language string) {
+	var pid string
+	assert.Eventually(s.T(),
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
You can use `require.Eventually` and remove the require at line 65

```suggestion
	require.Eventually(s.T(),
```

Reply:
Done!